aten/src/TH/vector/AVX2.cpp

```diff
@@ -105,18 +105,25 @@ void THFloatVector_normal_fill_AVX2(float *data,
   }
 }
 
-void THFloatVector_sigmoid_AVX2(float *y, const float *x, const ptrdiff_t n){
+void THFloatVector_sigmoid_AVX2(float *y, const float *x, const ptrdiff_t n) {
   ptrdiff_t i;
   const __m256 one = _mm256_set1_ps(1.0f);
-  __m256 YMM0, YMM1, YMM2;
-  for (i=0; i<=((n)-8); i+=8) {
-    YMM0 = _mm256_loadu_ps(x+i);
-    YMM1 = _mm256_add_ps(one, exp256_ps(YMM0));
-    YMM2 = _mm256_div_ps(one, YMM1);
-    _mm256_storeu_ps(y+i, YMM2);
+  const __m256 minus_one = _mm256_set1_ps(-1.0f);
+  __m256 YMM0, YMM1, YMM2, YMM3;
+  for (i = 0; i <= ((n)-16); i += 16) {
+    YMM0 = _mm256_loadu_ps(x + i);
+    YMM1 = _mm256_loadu_ps(x + i + 8);
+    YMM0 = _mm256_mul_ps(minus_one, YMM0);
```

Comment:
Is `_mm256_sub_ps(zero, YMM0)` or an xor of the floating point sign bit cheaper than a multiply here? Though it is probably not that important since exp256 dominates.

Reply:
Tried with `_mm256_sub_ps(zero, YMM0)`. Doesn't give any noticeable wins as exponent is the more expensive.